src/runtime/runtime_windows.go

```diff
@@ -0,0 +1,210 @@
+
+func postinit() {}
+
+//export mainCRTStartup
+func mainCRTStartup() int {
+	preinit()
+
+	// Obtain the initial stack pointer right before calling the run() function.
+	// The run function has been moved to a separate (non-inlined) function so
+	// that the correct stack pointer is read.
+	stackTop = getCurrentStackPointer()
+	runMain()
+
+	// For libc compatibility.
+	return 0
+}
+
+// Must be a separate function to get the correct stack pointer.
+//go:noinline
+func runMain() {
+	run()
+}
+
+var args []string
+
+//go:linkname os_runtime_args os.runtime_args
+func os_runtime_args() []string {
+	if args == nil {
+		// Obtain argc/argv from the environment.
+		_configure_narrow_argv(2)
+		argc := *__p___argc()
+		argv := *__p___argv()
+
+		// Make args slice big enough so that it can store all command line
+		// arguments.
+		args = make([]string, argc)
+
+		// Initialize command line parameters.
+		for i := 0; i < int(argc); i++ {
+			// Convert the C string to a Go string.
+			length := strlen(*argv)
+			arg := (*_string)(unsafe.Pointer(&args[i]))
+			arg.length = length
+			arg.ptr = (*byte)(*argv)
+			// This is the Go equivalent of "argv++" in C.
+			argv = (*unsafe.Pointer)(unsafe.Pointer(uintptr(unsafe.Pointer(argv)) + unsafe.Sizeof(argv)))
+		}
+	}
+	return args
+}
+
+func putchar(c byte) {
+	libc_putchar(int(c))
+}
+
+var heapSize uintptr = 128 * 1024 // small amount to start
+var heapMaxSize uintptr
+
+var heapStart, heapEnd uintptr
+
+func preinit() {
+	// Allocate a large chunk of virtual memory. Because it is virtual, it won't
+	// really be allocated in RAM. Memory will only be allocated when it is
+	// first touched.
+	heapMaxSize = 1 * 1024 * 1024 * 1024 // 1GB for the entire heap
+	const (
+		MEM_COMMIT     = 0x00001000
+		MEM_RESERVE    = 0x00002000
+		PAGE_READWRITE = 0x04
+	)
+	heapStart = uintptr(_VirtualAlloc(nil, heapMaxSize, MEM_COMMIT|MEM_RESERVE, PAGE_READWRITE))
+	heapEnd = heapStart + heapSize
+}
+
+type timeUnit int64
+
+var stackTop uintptr
+
+func ticksToNanoseconds(ticks timeUnit) int64 {
+	// Interrupt time count works in units of 100 nanoseconds.
+	return int64(ticks) * 100
+}
+
+func nanosecondsToTicks(ns int64) timeUnit {
+	// Interrupt time count works in units of 100 nanoseconds.
+	return timeUnit(ns) / 100
+}
+
+func sleepTicks(d timeUnit) {
+	// Calcluate milliseconds from ticks (which have a resolution of 100ns),
+	// rounding up.
+	milliseconds := int64(d+9_999) / 10_000
+	for milliseconds != 0 {
+		duration := uint32(milliseconds)
+		_Sleep(duration)
+		milliseconds -= int64(duration)
+	}
+}
+
+func ticks() timeUnit {
+	var unbiasedTime uint64
+	_QueryUnbiasedInterruptTime(&unbiasedTime)
+	return timeUnit(unbiasedTime)
+}
+
+//go:linkname now time.now
+func now() (sec int64, nsec int32, mono int64) {
+	var time uint64
+	_GetSystemTimeAsFileTime(&time)
+	sec = int64(time / 10_000_000)
+	nsec = int32((time - (uint64(sec) * 10_000_000)) * 100)
+	mono = ticksToNanoseconds(ticks())
+	return
+}
```

Comment:
I think it needs to be converted to Unix timestamp.

* time.Now()
    * tinygo run on windows : 2390/11/15 21:41:01
    * go run on windows : 2021/11/15 21:41:01

https://stackoverflow.com/questions/20370920/convert-current-time-from-windows-to-unix-timestamp-in-c-or-c

```suggestion
//go:linkname now time.now
func now() (sec int64, nsec int32, mono int64) {
	var time uint64
	_GetSystemTimeAsFileTime(&time)
	
	// convert from windows to unix time stamp
	time -= 0x019DB1DED53E8000
	
	sec = int64(time / 10_000_000)
	nsec = int32((time - (uint64(sec) * 10_000_000)) * 100)
	mono = ticksToNanoseconds(ticks())
	return
}
```

Reply:
Thanks! I will take a look.